Sources/StreamChatUI/ChatMessageList/ChatMessageListKeyboardObserver.swift

```diff
@@ -32,44 +29,32 @@ open class ChatMessageListKeyboardObserver {
         )
     }
     
-    public func unregister() {
+    open func unregister() {
         NotificationCenter.default.removeObserver(self)
     }
     
     @objc
-    private func keyboardWillChangeFrame(_ notification: Notification) {
+    open func keyboardWillChangeFrame(_ notification: Notification) {
         guard
             viewController?.presentedViewController == nil,
-            let userInfo = notification.userInfo,
-            let keyboardFrame = userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? CGRect,
-            let keyboardAnimationDuration = userInfo[UIResponder.keyboardAnimationDurationUserInfoKey] as? TimeInterval,
-            let keyboardAnimationCurve = userInfo[UIResponder.keyboardAnimationCurveUserInfoKey] as? UInt
+            let frame = (notification.userInfo?[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue,
+            let duration = notification.userInfo?[UIResponder.keyboardAnimationDurationUserInfoKey] as? Double,
+            let curve = notification.userInfo?[UIResponder.keyboardAnimationCurveUserInfoKey] as? UInt,
+            let containerView = containerView
         else { return }
 
-        let convertedKeyboardFrame = containerView.convert(keyboardFrame, from: UIScreen.main.coordinateSpace)
-        let intersectedKeyboardHeight = containerView.frame.intersection(convertedKeyboardFrame).height
```

Comment:
Deleting these lines will make the keyboard observer not supporting undocked keyboards on the iPad: More information in this WWDC talk: https://developer.apple.com/videos/play/wwdc2017/242/

Reply:
thanks for finding this 👍 put back in https://github.com/GetStream/stream-chat-swift/pull/1229